omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/PatientHeaderFragmentController.java

```diff
@@ -140,6 +143,24 @@ public void controller(FragmentConfiguration config, @SpringBean("emrApiProperti
             throw new APIException("Unable to generate name fields for patient header", e);
         }
 
+    }
+    
+    private Map<String,String> getAllNames(Patient patient) {
+    	Map<String,String> patientNames = getNames(patient.getPersonName());
+    	Set<PersonName> names = patient.getNames();
+    	
+    	// Get person names from person attribute types defined by 'extraPersonNames.personAttributeTypes' global property
+    	String personNameAttrTypeUuids = Context.getAdministrationService().getGlobalProperty("extraPersonNames.personAttributeTypes");
+    	if (StringUtils.isNotBlank(personNameAttrTypeUuids)) {
+    		for (String personNameAttrTypeUuid : personNameAttrTypeUuids.split(",")) {
+    			PersonAttributeType pat = Context.getPersonService().getPersonAttributeTypeByUuid(personNameAttrTypeUuid);
+    			PersonAttribute pa = patient.getAttribute(pat);
+    			if (pa != null) {
+    				patientNames.put(pat.getName(), pa.getValue());
```

Comment:
In order to somewhat support L10n you could do:
```java
uiUtils.message(pat.getName())
```

Reply:
This has been addressed